smstools/bugle.py

```diff
@@ -0,0 +1,59 @@
+import os, sys, time, sqlite3
+import core, sms_exceptions
+
+class Bugle:
+    """ New android (bugle_db) sqlite reader and writer """
+
+
+    def parse(self, filepath):
+        """ Parse a sqlite file to Text[] """
+
+        db = sqlite3.connect(filepath)
+        cursor = db.cursor()
+        texts = self.parse_cursor(cursor)
+        cursor.close()
+        db.close()
+        return texts
+
+    def parse_cursor(self, cursor):
+        texts = []
+        query = cursor.execute(
+            'select\
+            ppl.normalized_destination as num,\
+            p.timestamp as date,\
+            case when m.sender_id in (select _id from participants where contact_id=-1) then 2 else 1 end incoming,\
```

Comment:
By using -2 in the "where clause", this seems to partially work with my saved db.

```suggestion
            case when m.sender_id in (select _id from participants where contact_id=-2) then 2 else 1 end incoming,\
```
That's OK for conversations with known participants (in participants table).
For pushed SMS or unknown participants (without full_name in participants table), messages are assigned to the receiver instead of sender.

IMO, the SQL request needs to be refined to match all cases.

However, it was enough to save my life when my phone dropped me